.github/workflows/Test.yml

```diff
@@ -0,0 +1,79 @@
+name: Test
+
+on:
+  push:
+  pull_request:
+
+env:
+  GO111MODULE: on
+
+jobs:
+
+
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+        - { icon: 🐧, os: ubuntu }
+        - { icon: 🍎, os: macOS }
+        go:
+        - 1.14.x
+        - 1.15.x
+    name: '${{ matrix.platform.icon }} ${{ matrix.go }}'
+    runs-on: ${{ matrix.platform.os }}-latest
+    steps:
+
+    - name: '⚙️ Setup go'
+      uses: actions/setup-go@v1
+      with:
+        go-version: ${{ matrix.go }}
+
+    - name: '🧰 Checkout'
+      uses: actions/checkout@v2
+
+    - name: '⚙️ Install deps'
+      run: |
+        curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | bash -s -- -b $HOME/go/bin/ latest
+        go install github.com/kyoh86/richgo
+        go install github.com/mitchellh/gox
+
+    - name: '🚧 Run tests'
+      run: PATH=$HOME/go/bin/:$PATH make
+
+
+  test-win:
+    name: '🟪 MINGW64'
+    defaults:
+      run:
+        shell: msys2 {0}
+    runs-on: windows-latest
+    steps:
+
+    - name: '⚙️ Git config'
+      shell: bash
+      run: git config --global core.autocrlf input
+
+    - name: '🟪 Setup MSYS2'
+      uses: msys2/setup-msys2@v2
```

Comment:
We should use the official virtual environments: https://github.com/actions/virtual-environments

Reply:
We are using the official virtual environments (`ubuntu-latest`, `macos-latest` and `windows-latest`). MSYS2 is preinstalled in `windows-latest`, and Action setup-msys2 allows to either set it up, or get a clean tarball and extract it. As explained in https://github.com/msys2/setup-msys2#context, the problem with the built-in setup is that up to 10-12min update times might be required. That's why the recommended approach is using the clean tarball (with caching enabled by default).

My perception is that you mean to use cmd/powershell and not to rely on MSYS2. I can understand that wish, but AFAICT cobra was never tested on Windows. Moreover, the build/testing plumbing depends on Unix tooling which is not available on Windows by default. Hence, reworking the testing approach for supporting a regular Windows environment is out of the scope of this PR. It can be added as an additional job/matrix afterwards.

Reply:
I see - thanks for the detailed explanation! And that makes sense. If this provides a clean, cached tar for us to use in the action, sounds good to me. I also looked over the repo and the Github org and seems 👍

Reply:
In fact, there is feedback between MSYS2 and GitHub: https://github.com/actions/virtual-environments/pull/2395#issuecomment-757546669. At some point, the action might be merged/copied in the default environment.